content/shmem_wait_until.tex

```diff
@@ -68,6 +64,11 @@
     to \FUNC{shmem\_wait\_until} blocks until the value of \VAR{ivar} at the
     calling \ac{PE} satisfies the wait condition specified by the comparison
     operator, \VAR{cmp}, and comparison value, \VAR{cmp\_value}.
+
+    Implementations must ensure that \FUNC{shmem\_wait} and
+    \FUNC{shmem\_wait\_until} do not return before the update of the memory
+    indicated by \VAR{ivar} is fully complete.  Partial updates to the memory
+    must not cause \FUNC{shmem\_wait} or \FUNC{shmem\_wait\_until} to return.
 }
```

Comment:
As we have discussed, this might require (1) wait/test to be implemented as a atomic operation and (2) shmem_p to be implemented as atomic operation

Reply:
This ticket excludes `shmem_p` from usage with the wait/test APIs.  So (2) is a non-issue.

In WG discussion on (1), we have determined that wait/test should not observe partial updates to memory (including updates made by AMOs, as noted in your comment above).  This is required by existing applications.  I agree with the observations that (a) this is not easy to implement and (b) overheads can be lower in implementations that allow partial updates.  As discussed in the WG, these are good arguments for introducing a new, more portable point-to-point synchronization API.  In this ticket, we are constraining ourselves to semantic clarifications that continue to support existing usage models.

If you don't mind, I'll ping you separately for a discussion on how to close this issue.